test/colony.js

```diff
@@ -180,6 +199,86 @@ contract("Colony", () => {
     });
   });
 
+  describe("when bootstrapping the colony", () => {
+    const INITIAL_REPUTATIONS = [toBN(5 * 1e18).toString(), toBN(4 * 1e18).toString(), toBN(3 * 1e18).toString(), toBN(2 * 1e18).toString()];
+    const INITIAL_ADDRESSES = addresses.slice(0, 4);
+
+    it("should assign reputation correctly when bootstrapping the colony", async () => {
+      const skillCount = await colonyNetwork.getSkillCount.call();
+
+      await colony.mintTokens(toBN(14 * 1e18).toString());
+      await colony.bootstrapColony(INITIAL_ADDRESSES, INITIAL_REPUTATIONS);
+
+      const numberOfReputationLogs = await colonyNetwork.getReputationUpdateLogLength(true);
+      assert.equal(numberOfReputationLogs.toNumber(), INITIAL_ADDRESSES.length);
+
+      const updateLog = await colonyNetwork.getReputationUpdateLogEntry(0, true);
+      assert.equal(updateLog[0], INITIAL_ADDRESSES[0]);
+      assert.equal(updateLog[1].toString(), INITIAL_REPUTATIONS[0]);
+      assert.equal(updateLog[2].toString(), skillCount.toNumber());
+    });
+
+    it("should assign tokens correctly when bootstrapping the colony", async () => {
+      await colony.mintTokens(toBN(14 * 1e18).toString());
+      await colony.bootstrapColony(INITIAL_ADDRESSES, INITIAL_REPUTATIONS);
+
+      const balance = await token.balanceOf(INITIAL_ADDRESSES[0]);
+      assert.equal(balance.toString(), INITIAL_REPUTATIONS[0]);
+    });
+
+    it("should be able to bootstrap colony more than once", async () => {
+      const amount = toBN(10 * 1e18).toString();
+      await colony.mintTokens(amount);
+      colony.bootstrapColony([INITIAL_ADDRESSES[0]], [INITIAL_REPUTATIONS[0]]);
+      colony.bootstrapColony([INITIAL_ADDRESSES[0]], [INITIAL_REPUTATIONS[0]]);
+
+      const balance = await token.balanceOf(INITIAL_ADDRESSES[0]);
+      assert.equal(balance.toString(), amount);
+    });
+
+    it("should throw if length of inputs is not equal", async () => {
```

Comment:
Strictly, this test is only checking it throws if `INITIAL_ADDRESSES` is longer then `INITIAL REPUTATIONS`. If the reverse is true, it's not going to throw. Either add a check for the inputs being the same length or reword this test's description. I think I have a slight preference for the former; if the inputs are not the same, it's almost certainly a mistake on the user's part and so we shouldn't take action that they can't revert.

Reply:
Agreed, thank you.